### PR TITLE
Implement IO#initialize_copy

### PR DESCRIPTION
**Pull request by ksss**
This is ported PR from https://github.com/iij/mruby-io/pull/27

### CODE REVIEW

File: mrbgems/mruby-io/src/io.c [hunk 1]
```diff
@@ -530,6 +530,54 @@ mrb_io_s_popen(mrb_state *mrb, mrb_value klass)
...
+  struct mrb_io *fptr_copy;
+  struct mrb_io *fptr_orig;
+
+  mrb_get_args(mrb, "o", &orig);
+  fptr_copy = (struct mrb_io *)DATA_PTR(copy);
+  if (fptr_copy != NULL) {
+    fptr_finalize(mrb, fptr_copy, FALSE);
+    mrb_free(mrb, fptr_copy);
+  }
+  fptr_copy = (struct mrb_io *)mrb_io_alloc(mrb);
```
Comment (tsahara) [inline mrbgems/mruby-io/src/io.c]: `fptr_copy` will not be freed (causes memory leakage) when an exception is raised on 562-576.

File: mrbgems/mruby-io/src/io.c [hunk 2]
```diff
@@ -530,6 +530,54 @@ mrb_io_s_popen(mrb_state *mrb, mrb_value klass)
...
+  fptr_orig = io_get_open_fptr(mrb, orig);
+
+  buf = mrb_iv_get(mrb, orig, mrb_intern_cstr(mrb, "@buf"));
+  mrb_iv_set(mrb, copy, mrb_intern_cstr(mrb, "@buf"), buf);
+
+  fptr_copy->fd = mrb_dup(mrb, fptr_orig->fd);
```
Comment (tsahara) [inline mrbgems/mruby-io/src/io.c]: descriptor leakage here (`mrb_dup` on 568 may raise an exception)

**Comment by matz** [comment]
@tsahara I think I addressed the leakage issues.

**Comment by ksss** [comment]
Thank you for reviewing and fixing.

**Comment by tsahara** [comment]
@matz The fixes look good.  Thanks.

**Pull request merged**